app/src/org/commcare/views/widgets/AudioPrototype.java

```diff
@@ -0,0 +1,211 @@
+import android.app.Activity;
+import android.content.ActivityNotFoundException;
+import android.content.Context;
+import android.content.Intent;
+import android.media.MediaPlayer;
+import android.net.Uri;
+import android.support.v4.app.FragmentActivity;
+import android.support.v4.app.FragmentManager;
+import android.view.LayoutInflater;
+import android.view.View;
+import android.widget.ImageButton;
+import android.widget.LinearLayout;
+import android.widget.TextView;
+import android.widget.Toast;
+
+import org.commcare.activities.FormEntryActivity;
+import org.commcare.dalvik.R;
+import org.commcare.logging.analytics.GoogleAnalyticsUtils;
+import org.commcare.logic.PendingCalloutInterface;
+import org.commcare.utils.StringUtils;
+import org.javarosa.core.services.locale.Localization;
+import org.javarosa.form.api.FormEntryPrompt;
+
+/**
+ * Created by Saumya on 6/3/2016.
+ * An alternative audio widget that records and plays audio natively without callout to any external application
+ */
+public class AudioPrototype extends AudioWidget implements RecordingFragment.RecordingCompletionListener {
+
+    private RecordingFragment recorder;
+    private FragmentManager fm;
+    private LinearLayout layout;
+    private ImageButton mPlayButton;
+    private TextView recordingNameText;
+
+    public AudioPrototype(Context context, FormEntryPrompt prompt, PendingCalloutInterface pic){
+        super(context, prompt, pic);
+        fm = ((FragmentActivity) getContext()).getSupportFragmentManager();
+        recorder = new RecordingFragment();
+        recorder.setListener(this);
+    }
+
+    @Override
+    protected void initializeButtons(final FormEntryPrompt prompt){
+        LayoutInflater vi = (LayoutInflater)getContext().getSystemService(Context.LAYOUT_INFLATER_SERVICE);
+        layout = (LinearLayout) vi.inflate(R.layout.audio_prototype, null);
+
+        mPlayButton = (ImageButton) layout.findViewById(R.id.play_audio);
+        ImageButton mCaptureButton = (ImageButton) layout.findViewById(R.id.capture_button);
+        ImageButton mChooseButton = (ImageButton) layout.findViewById(R.id.choose_file);
+
+        mCaptureButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                GoogleAnalyticsUtils.reportRecordingPopupOpened();
+                captureAudio(prompt);
+            }
+        });
+
+        // launch audio filechooser intent on click
+        mChooseButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                GoogleAnalyticsUtils.reportAudioFileChosen();
+                Intent i = new Intent(Intent.ACTION_GET_CONTENT);
+                i.setType("audio/*");
+                try {
+                    ((Activity)getContext()).startActivityForResult(i, FormEntryActivity.AUDIO_VIDEO_FETCH);
+                    recordingNameText.setTextColor(getResources().getColor(R.color.black));

```

Comment:
If you change it to `getColor(R.color.black, null)` you should be good. While in some cases it isn't a huge deal to call deprecated code, it is generally good to avoid it. Calls to the Android platform can break in really crazy ways and sometimes that is due to using deprecated calls, so always good to rule it out when possible.